Specs/Scene/createTileMapServiceImageryProviderSpec.js

```diff
@@ -254,6 +254,23 @@ defineSuite([
         });
     });
 
+    it('resource request takes a query string', function() {
+        /*jshint unused: false*/
+        var requestMetadata = when.defer();
+        spyOn(loadWithXhr, 'load').and.callFake(function(url, responseType, method, data, headers, deferred, overrideMimeType) {
+            requestMetadata.resolve(url);
+            deferred.reject(); //since the TMS server doesn't exist (and doesn't need too) we can just reject here.
+        });
+
+        var provider = createTileMapServiceImageryProvider({
+            url : 'server.invalid?query=1',
+        });
+
+        return requestMetadata.promise.then(function(url) {
+            expect(url.indexOf('?query=1')).not.toEqual(-1);
```

Comment:
I believe this test still passes without your bugfix, since `?query=1` was still in the string, just not at the end.  instead of indexOf, use `/\?query=1$/.test(url)` and expect it to be true.